Bottom: e78599926da20ccdbd06ff93d74ad6cfae58dcb5
Top:    6bcf73039854ff80a580942bf1a5c9be654d5f75
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-12-10 01:19:10 +0000

Add split colouring for "Find" action, introducing new flag "connected"

Colour differently depending on whether objects are physically connected
to the object under the cursor when invoking the "Find" action.

The aim here is to differentiate between what an ohm-meter would show
connected with the board as-currently designed, and what it would read
if all rat-lines were implemented.


---

diff --git a/src/action.c b/src/action.c
index 189e020..bfafd6f 100644
--- a/src/action.c
+++ b/src/action.c
@@ -2282,12 +2282,13 @@ ActionConnection (int argc, char **argv, Coord x, Coord y)
 	case F_Find:
 	  {
 	    gui->get_coords (_("Click on a connection"), &x, &y);
-	    LookupConnection (x, y, true, 1, FOUNDFLAG, false);
+	    LookupConnection (x, y, true, 1, CONNECTEDFLAG, false);
+	    LookupConnection (x, y, true, 1, FOUNDFLAG, true);
 	    break;
 	  }
 
 	case F_ResetLinesAndPolygons:
-	  if (ClearFlagOnLinesAndPolygons (true, FOUNDFLAG))
+	  if (ClearFlagOnLinesAndPolygons (true, CONNECTEDFLAG | FOUNDFLAG))
 	    {
 	      IncrementUndoSerialNumber ();
 	      Draw ();
@@ -2295,7 +2296,7 @@ ActionConnection (int argc, char **argv, Coord x, Coord y)
 	  break;
 
 	case F_ResetPinsViasAndPads:
-	  if (ClearFlagOnPinsViasAndPads (true, FOUNDFLAG))
+	  if (ClearFlagOnPinsViasAndPads (true, CONNECTEDFLAG | FOUNDFLAG))
 	    {
 	      IncrementUndoSerialNumber ();
 	      Draw ();
@@ -2303,7 +2304,7 @@ ActionConnection (int argc, char **argv, Coord x, Coord y)
 	  break;
 
 	case F_Reset:
-	  if (ClearFlagOnAllObjects (true, FOUNDFLAG))
+	  if (ClearFlagOnAllObjects (true, CONNECTEDFLAG | FOUNDFLAG))
 	    {
 	      IncrementUndoSerialNumber ();
 	      Draw ();
